File: flexizen-backend/src/main/java/com/flexizen/service/PageService.java
package com.flexizen.service;

import com.flexizen.model.PageContent;
import com.flexizen.repository.PageContentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class PageService {

    @Autowired
    private PageContentRepository pageContentRepository;

    public PageContent getPageContent(String pageType) {
        return pageContentRepository.findByPageType(pageType.toUpperCase());
    }

    public PageContent updatePageContent(String pageType, String content) {
        String type = pageType.toUpperCase();
        PageContent page = pageContentRepository.findByPageType(type);
        if (page == null) {
            page = new PageContent();
            page.setPageType(type);
        }
        page.setContent(content);
        pageContentRepository.save(page);
        return page;
    }
}

File: flexizen-frontend/src/services/pageService.js
import api from './api';

export const pageService = {
    // Public API
    getPageContent: async (pageType) => {
        const response = await api.get(`/pages/${pageType}`);
        return response.data;
    },

    // Admin API
    updatePageContent: async (pageType, content) => {
        const response = await api.put(`/admin/pages/${pageType}`, { content });
        return response.data;
    }
};
